x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.js

```diff
@@ -43,31 +37,28 @@ export class MapsAppView extends React.Component {
   _globalSyncChangeMonitorSubscription = null;
   _appSyncUnsubscribe = null;
   _appStateManager = new AppStateManager();
+  _prevIndexPatternIds = null;
 
   constructor(props) {
     super(props);
     this.state = {
       indexPatterns: [],
-      prevIndexPatternIds: [],
       initialized: false,
       savedQuery: '',
       initialLayerListConfig: null,
     };
   }
 
   componentDidMount() {
-    // Init sync utils
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    this._globalSyncUnsubscribe = useGlobalStateSyncing();
-    // eslint-disable-next-line react-hooks/rules-of-hooks
-    this._appSyncUnsubscribe = useAppStateSyncing(this._appStateManager);
+    this._isMounted = true;
+
+    this._globalSyncUnsubscribe = startGlobalStateSyncing();
+    this._appSyncUnsubscribe = startAppStateSyncing(this._appStateManager);
     this._globalSyncChangeMonitorSubscription = getData().query.state$.subscribe(
       this._updateFromGlobalState
     );
 
-    // Check app state in case of refresh
     const initAppState = this._appStateManager.getAppState();
-    this._onQueryChange(initAppState);
     if (initAppState.savedQuery) {
```

Comment:
This `if` condition could be removed while in here. It's checked in `_updateStateFromSavedQuery`